<?xml version="1.0"?>

<launch>

    <!-- support multi robot -->
    <arg name="robot_name" default=""/>   <!-- support multi robot -->
    <!-- robot frame -->
    <arg name="base_frame"       default="base_footprint" />    
    <arg name="odom_frame"       default="odom" />
    <arg name="imu_frame"       default="imu" /> 
    <!-- pub topic -->
    <arg name="odom_topic"       default="odom" /><!-- do NOT use '/' in topic name-->
    <arg name="imu_topic"       default="imu" />
    <arg name="battery_topic"       default="battery" />
    <!-- sub topic -->
    <arg name="cmd_vel_topic"       default="cmd_vel" /> <!-- do NOT use '/' in topic name-->
    <arg name="ackermann_cmd_topic"       default="ackermann_cmd" />
    <!-- config param -->
    <arg name="pub_imu"       default="False" />
    <arg name="sub_ackermann"       default="False" />
    <arg name="pub_sonar"       default="False" />
    <!-- serial param -->
    <arg name="port" default="/dev/ttyACM0" />
    <arg name="baudrate" default="57600" />


    <!-- serial node node -->
    <node name="serial_node" pkg="robot_navigation" type="serial_node.py" output="screen">
        <param name="port" value='$(arg port)' />
        <param name="baudrate" value="$(arg baudrate)" />
    </node>

    <!-- base control node -->
    <!--ros namespace name can NOT be empty,so need evaluate use multi robot or not-->
    <node name="test_base_control"   pkg="robot_navigation"  type="test_base_control.py" output="screen">
            <!--param name="baudrate"   value="115200"/>  
            <param name="port"  value='/dev/move_base'/-->
            <param name="port" value='$(arg port)' />
            <param name="baudrate" value="$(arg baudrate)" />

            <param name="base_id"    value="$(arg base_frame)"/> <!-- base_link name -->
            <param name="odom_id"    value="$(arg odom_frame)"/>      <!-- odom link name -->
            <param name="imu_id"    value="$(arg imu_frame)"/>      <!-- imu link name -->

            <param name="odom_topic" value="$(arg odom_topic)"/> <!-- topic name -->
            <param name="imu_topic" value="$(arg imu_topic)"/> <!-- topic name -->
            <param name="battery_topic" value="$(arg battery_topic)"/> <!-- topic name -->

            <param name="cmd_vel_topic" value="$(arg cmd_vel_topic)"/> 
            <param name="ackermann_cmd_topic" value="$(arg ackermann_cmd_topic)"/> <!-- topic name -->

            <param name="pub_imu" value="$(arg pub_imu)"/> <!-- pub imu topic or not -->
            <param name="pub_sonar" value="$(arg pub_sonar)"/> <!-- pub sonar topic or not -->
            
            <param name="sub_ackermann" value="$(arg sub_ackermann)"/> <!-- sub ackermann topic or not -->
    </node>
    
</launch>
